tests/Integration/Behaviour/Features/Scenario/Product/generate_combinations.feature

```diff
@@ -0,0 +1,36 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s product --tags generate-combinations
+@reset-database-before-feature
+@clear-cache-before-feature
+@generate-combinations
+Feature: Generate attribute combinations for product in Back Office (BO)
+  As an employee
+  I need to be able to generate product attribute combinations from BO
+
+  Background:
+    Given language with iso code "en" is the default one
+    And attribute group "size" named "Size" in en language exists
+    And attribute group "color" named "Color" in en language exists
+    And attribute "s" named "S" in en language exists
+    And attribute "m" named "M" in en language exists
+    And attribute "l" named "L" in en language exists
+    And attribute "white" named "White" in en language exists
+    And attribute "black" named "Black" in en language exists
+    And attribute "blue" named "Blue" in en language exists
+    And attribute "red" named "Red" in en language exists
+
+  Scenario: Generate product combinations
+    When I add product "product1" with following information:
+      | name       | en-US:universal T-shirt |
+      | is_virtual | false                   |
+    Then product product1 type should be standard
+    When I generate combinations for product product1 using following attributes:
+      | size       | [s,m]               |
+      | color      | [white,black,blue]  |
+    Then product product1 should have following combinations:
+      | combination name                 |
+      | Size - S, Color - White          |
+      | Size - S, Color - Black          |
+      | Size - S, Color - Blue           |
+      | Size - M, Color - White          |
+      | Size - M, Color - Black          |
+      | Size - M, Color - Blue           |
```

Comment:
Could you add a test for pagination please? In another scenario maybe where you would get Page 1 of 3 and Page 2 of 3

Reply:
The total combinations should also be tested

Reply:
And the list of attributes is not tested either Maybe you could add a behat function to check:

```
And (first|second) combination of product product1 should have the following attributes:
    | size   | S        |
    | color | white |
```

For the offset either use plain words like what was done for invoice behat tests:
https://github.com/PrestaShop/PrestaShop/blob/c020552ca5d94aa1637fb8af6ccb37340c2c95be/tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php#L569

It's just a matching between word and index, it doesn't allow to check all positions but for the limited test cases it was enough

If you really want to be able to check any index just use an int with sentence `combination at position 12` or we could add a method in `PrimitiveUtils` to handle this

Reply:
I actually discovered `PrimitiveUtils::castStringIntegerIntoInteger` existed so you can use this one, and maybe it could be improved to allow a little parsing of  `11th` `18th` `21st` `22nd` `23rd` (I believe there are only those four suffixes)